pkg/storage/iterators.go

```diff
@@ -199,3 +199,67 @@ func NewFilteredTupleKeyIterator(iter TupleKeyIterator, filter TupleKeyFilterFun
 		filter,
 	}
 }
+
+// TupleKeyConditionFilterFunc is a filter function that is used to filter out
+// tuples from a [TupleKeyIterator] that don't meet the tuple the conditions provided by the request.
+// Implementations should return true if the tuple should be returned
+// and false if it should be filtered out.
+// errors will be treated as false. The iterator can return the last seen error before ErrIteratorDone.
```

Comment:
This comment doesn't seem accurate - if you return 5 tuples, 1 of which has an EvaluationError, and then it's the end of the iterator, you will evaluate line 229, skip it, and return `ErrIteratorDone`, no?

Reply:
The behaviour was incorrect and is now fixed.  At the end of the iterator, it should return error.

Reply:
No unit tests?

Reply:
Unit tests added